### PR TITLE
Metrics upgrade

### PR DESCRIPTION
**Pull request by sjezewski**
Ok, ready for review!

This PR polls the database to get the counts of different objects in the cluster to report to segment. It replaces the old method of using atomic counters.

### CODE REVIEW

File: src/server/pkg/metrics/metrics.go [hunk 1]
```diff
@@ -1,68 +1,56 @@
...
-func AddJobs(num int64) {
-	atomic.AddInt64(&metrics.Jobs, num)
-	atomic.SwapInt64(&modified, 1)
-}
-
-// AddPipelines atomically adds a number of pipelines to metrics.
-func AddPipelines(num int64) {
-	atomic.AddInt64(&metrics.Pipelines, num)
-	atomic.SwapInt64(&modified, 1)
+func latestMetricsFromDB(dbClient *gorethink.Session, pfsDbName string, ppsDbName string, metrics *Metrics) {
```
Comment (jdoliner) [inline src/server/pkg/metrics/metrics.go]: Small nit here, this could just be called `dbMetrics` to match `externalMetrics`.

**Comment by jdoliner** [comment]
Cool this all looks reasonable / correct to me.

File: src/server/pkg/metrics/metrics.go (hunk 1 above)
Comment (sjezewski) [inline src/server/pkg/metrics/metrics.go]: Cool, change made

**Comment by jdoliner** [comment]
sam-lgtm

**Pull request merged**